go-controller/pkg/ovn/external_ids_syncer/port_group/port_group_sync.go

```diff
@@ -0,0 +1,230 @@
+	return libovsdbops.NewDbObjectIDs(libovsdbops.PortGroupNamespace, controllerName, map[libovsdbops.ExternalIDKey]string{
+		libovsdbops.ObjectNameKey: namespace,
+	})
+}
+
+func getPortGroupNetpolNamespaceDbIDs(namespace, direction, networkExternalID string) *libovsdbops.DbObjectIDs {
+	controllerName := getControllerName(networkExternalID)
+	return libovsdbops.NewDbObjectIDs(libovsdbops.PortGroupNetpolNamespace, controllerName, map[libovsdbops.ExternalIDKey]string{
+		libovsdbops.ObjectNameKey:      namespace,
+		libovsdbops.PolicyDirectionKey: direction,
+	})
+}
+
+func getPortGroupNetworkPolicyDbIDs(policyNamespace, policyName, networkExternalID string) *libovsdbops.DbObjectIDs {
+	controllerName := getControllerName(networkExternalID)
+	return libovsdbops.NewDbObjectIDs(libovsdbops.PortGroupNetworkPolicy, controllerName,
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey: fmt.Sprintf("%s_%s", policyNamespace, policyName),
+		})
+}
+
+func getPortGroupClusterDbIDs(baseName, networkExternalID string) *libovsdbops.DbObjectIDs {
+	controllerName := getControllerName(networkExternalID)
+	return libovsdbops.NewDbObjectIDs(libovsdbops.PortGroupCluster, controllerName, map[libovsdbops.ExternalIDKey]string{
+		libovsdbops.ObjectNameKey: baseName,
+	})
+}
+
+// getReferencingObjsAndNewDbIDs finds all object that reference stale port group and tries to create a new dbIDs
+// based on referencing objects
+func (syncer *PortGroupSyncer) getReferencingObjsAndNewDbIDs(oldHash, oldName, networkExternalID string) (acls []*nbdb.ACL,
+	dbIDs *libovsdbops.DbObjectIDs, err error) {
+	// find all referencing objects
+	aclPred := func(acl *nbdb.ACL) bool {
+		return strings.Contains(acl.Match, "@"+oldHash)
+	}
+	acls, err = libovsdbops.FindACLsWithPredicate(syncer.nbClient, aclPred)
+	if err != nil {
+		err = fmt.Errorf("failed to find acls for port group %s: %v", oldHash, err)
+		return
+	}
+	// build dbIDs
+	switch {
+	// Filter port groups with pre-defined names
+	case oldName == types.ClusterPortGroupNameBase || oldName == types.ClusterRtrPortGroupNameBase:
+		dbIDs = getPortGroupClusterDbIDs(oldName, networkExternalID)
+	case strings.Contains(oldName, "_"):
+		// network policy owned namespace
+		s := strings.SplitN(oldName, "_", 2)
+		if s[1] == ingressDefaultDenySuffix || s[1] == egressDefaultDenySuffix {
+			// default deny port group, name format = hash(namespace)_gressSuffix
+			// need to find unhashed namespace name, use referencing ACLs
+			if len(acls) == 0 {
+				// default deny port group should always have acls
+				err = fmt.Errorf("defaultDeny port group doesn't have any referencing ACLs, can't extract namespace")
+				return
+			}
+			// all default deny acls will have the same namespace as ExternalID
+			acl := acls[0]
+			namespace := acl.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+			var direction string
+			if s[1] == ingressDefaultDenySuffix {
+				direction = "Ingress"
+			} else {
+				direction = "Egress"
```

Comment:
dont we have constants defined for these?

Reply:
I don't use any external constants in this package on purpose: so in case we change them for "current" code, sync code stays independent and focused on a pre-defined db format parsing

Reply:
Was not really or only because of the strings in code thingy, but to have a good place to explain through a comment why you are defining constants that are already defined elsewhere. This is still missing but this is still a nit :P